hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledTaskDescriptor.java

```diff
@@ -98,22 +103,24 @@ void setState(Map<?, ?> snapshot) {
 
     void setScheduledFuture(ScheduledFuture<?> future) {
         this.future = future;
+        this.status.set(ACTIVE);
     }
 
-    void setTaskResult(ScheduledTaskResult result) {
-        this.resultRef.set(result);
+    boolean isActive() {
+        return this.status.get() == ACTIVE;
```

Comment:
Generally speaking, what if we didn't have this additional status and we relied on the results of the old `ScheduledTaskDescriptor#canBeScheduled` implementation?

Reply:
Will merge this PR. There is an on-going discussion on slack regarding this, based on the outcome we can open another PR to address it if needed.